docs/adr/adr-002-predevnet-core-to-full-communication.md

```diff
@@ -0,0 +1,39 @@
+# ADR for Devnet Celestia Core <> Celestia Node Communication 
+
+## Authors
+
+@renaynay
```

Comment:
Kindly ask to add me as a coauthor. Even I didn't write a line in this doc, the core idea of embedding a node and having two modes was mine, as per [this](https://github.com/celestiaorg/celestia-node/pull/48#discussion_r702846034) comment and one of our sync discussions.